runtime/parser/declaration.go

```diff
@@ -1228,6 +1229,25 @@ func parseMemberOrNestedDeclaration(p *parser, docString string) (ast.Declaratio
 			p.next()
 			continue
 
+		case lexer.TokenPragma:
+			if previousIdentifierToken != nil {
+				return nil, NewSyntaxError(
+					previousIdentifierToken.StartPos,
+					"unexpected token: %s",
+					previousIdentifierToken.Type,
+				)
+			}
+			if access != ast.AccessNotSpecified {
+				return nil, NewSyntaxError(*accessPos, "invalid access modifier for pragma")
+			}
+			if staticModifierEnabled && staticPos != nil {
+				return nil, NewSyntaxError(*staticPos, "invalid static modifier for pragma")
+			}
+			if nativeModifierEnabled && nativePos != nil {
+				return nil, NewSyntaxError(*nativePos, "invalid native modifier for pragma")
+			}
+			return parsePragmaDeclaration(p)
```

Comment:
This part is identical to the top-level pragma parsing: https://github.com/onflow/cadence/blob/6ed330eb763401e446f37e38df235685dc0cc41e/runtime/parser/declaration.go#L80-L89

Maybe refactor it into a function and re-use

Reply:
Good idea! Refactored it for pragma and other declarations in 2dc40915eeea7dae8b51f5431ccb5cec19525931